_posts/2017-07-23-fastlane-travis-weekly-deploys.md

```diff
@@ -0,0 +1,314 @@
+
+<!-- more -->
+
+<center>
+  <img src="/images/fastlane-weekly/screenshot.png" width=300>
+</center>
+
+As the JavaScript is continuously deployed, the native side of the app rarely gets a deploy. In order to ensure an up-to-date version of the app, I used the [scheduler][schedule] now available in Travis CI, and Circle CI. This is a perfect use-case for one-off tasks like uploading an app to Apple's Testflight on a weekly basis.
+
+I wanted this to exist outside of our current CI environment for two reasons:
+
+* Our CI is already [using AppHub][pr_deploy] to deploy the JavaScript parts of our React Native on a per-commit basis. It's complicated enough as it is, without adding a lot more process.
+* Our CI is currently running on Linux boxes, and so everything is fast and stable. Deploying using the main repo would force us to use macOS which would slow down our processes.
+
+The downside of this choice is that the process of uploading is not inside the main repo, and can go out of sync with the main app.
+
+## Setup
+
+I created a new repo, and added the [usual LICENSE and README][init], then started [working on a PR][pr] that added support for CI to run. Here are the general steps I needed to make work:
+
+* Downloading and setting up the application.
+* Ensuring signing will work.
+* Creating the build and shipping it to Testflight.
+* Notifications that it passed or succeeded.
+
+Finally I needed to document the process, which is what you're reading.
+
+## Downloading and setting up the application
+
+My initial thoughts were to use a submodule, but that option provides little advantage over cloning the repo itself so it's done inline. Our dependencies for the app live in Rubygems (Fastlane/CocoaPods), NPM (React Native) and CocoaPods (Artsy Mobile code), so I use the `before_install` and `before_script` section of the `.travis.yml` to set up our dependencies:
+
+```yml
+# Use a Mac build please
+language: objective-c
+osx_image: xcode8.2
+
+# Ensure that Fastlane is at the latest version
+before_install:
+- bundle update
+
+# Let fastlane set up the other dependency managers
+before_script:
+- bundle exec fastlane setup
+
+# Separate fastlane lanes so that they can be individually
+# tested one by one during development
+script:
+- bundle exec fastlane validate_env_vars
+- bundle exec fastlane stamp_plist
+- bundle exec fastlane ship
+```
+
+Note the `- bundle update`. As fastlane works against iTunes connect which is always changing locking the version doesn't make much sense. 
+
+## Ensuring signing will work.
+
+This one is a bit tricker, luckily I've already set up one of our apps to use [fastlane match][] and I can re-use that infrastructure. As it is a private repo, Travis did not have access to clone the repo. I fixed this by creating an access token for a user with read-only access to our match-codesigning repo, then exposed this as a private environment variable in CI which the Matchfile uses. E.g.
```

Comment:
I think the Travis CI logging is covering that problem, but Circle didnt

![screen shot 2017-07-28 at 13 09 14](https://user-images.githubusercontent.com/49038/28728277-090589ac-7396-11e7-8882-af02c35237ee.png)

IMO, it's hard for _you_ to know what is a private ENV var or not. Perhaps you could argue _all_ external ENV vars could be redacted. Hard to say.